File: tcbn/java/create/tcbn_create.java
// -------------------------------------------------------------------
//
//	java/create/tcbn_create.java
//
//						Jun/16/2011
// -------------------------------------------------------------------
import tokyocabinet.BDB;

// -------------------------------------------------------------------
public class tcbn_create
{
// -------------------------------------------------------------------
public static void main (String[] args)
{
System.out.println ("*** 開始 ***");

	BDB bdb = new BDB();

	String file_in = "/var/tmp/tokyo_cabinet/cities.tcb";

	if(!bdb.open (file_in, BDB.OWRITER | BDB.OCREAT))
		{
		int ecode = bdb.ecode();
		System.err.println("open error: " + bdb.errmsg(ecode));
		}

	tcbn_manipulate.insert_proc (bdb,"t0831","水戸",71235,"2008-8-10");
	tcbn_manipulate.insert_proc (bdb,"t0832","古河",65147,"2008-9-11");
	tcbn_manipulate.insert_proc (bdb,"t0833","結城",41268,"2008-10-14");
	tcbn_manipulate.insert_proc (bdb,"t0834","土浦",93591,"2008-3-24");
	tcbn_manipulate.insert_proc (bdb,"t0835","牛久",72746,"2008-5-18");
	tcbn_manipulate.insert_proc (bdb,"t0836","取手",32189,"2008-4-17");
	tcbn_manipulate.insert_proc (bdb,"t0837","つくば",84265,"2008-1-30");
	tcbn_manipulate.insert_proc (bdb,"t0838","日立",65973,"2008-2-12");
	tcbn_manipulate.insert_proc (bdb,"t0839","守谷",95146,"2008-9-24");

	tcbn_manipulate.display_proc (bdb);

	if(!bdb.close())
		{
		int ecode = bdb.ecode();
		System.err.println("close error: " + bdb.errmsg(ecode));
		}

System.out.println ("*** 終了 ***");
}
// -------------------------------------------------------------------

// -------------------------------------------------------------------
}
// -------------------------------------------------------------------

File: tcbn/scala/create/tcbn_create.scala
// -------------------------------------------------------------------
//
//	create/tcbn_create.scala
//
//						Jun/16/2011
// -------------------------------------------------------------------
import tokyocabinet.BDB

// -------------------------------------------------------------------
object tcbn_create
{
// -------------------------------------------------------------------
def main (args: Array[String])
{
	println ("*** 開始 ***")

	val bdb = new BDB()

	val file_in = "/var/tmp/tokyo_cabinet/cities.tcb"

	if(!bdb.open (file_in, BDB.OWRITER | BDB.OCREAT))
		{
		val ecode = bdb.ecode()
//		println("open error: " + bdb.errmsg(ecode))
		println("open error: ")
		}

	tcbn_manipulate.insert_proc (bdb,"t0831","水戸",32581,"1998-4-10")
	tcbn_manipulate.insert_proc (bdb,"t0832","古河",65147,"1998-7-11")
	tcbn_manipulate.insert_proc (bdb,"t0833","結城",48256,"1998-10-14")
	tcbn_manipulate.insert_proc (bdb,"t0834","土浦",91543,"1998-3-24")
	tcbn_manipulate.insert_proc (bdb,"t0835","牛久",72739,"1998-5-12")
	tcbn_manipulate.insert_proc (bdb,"t0836","取手",39125,"1998-4-17")
	tcbn_manipulate.insert_proc (bdb,"t0837","つくば",87246,"1998-1-30")
	tcbn_manipulate.insert_proc (bdb,"t0838","日立",65937,"1998-2-12")
	tcbn_manipulate.insert_proc (bdb,"t0839","守谷",14698,"1998-5-24")

	tcbn_manipulate.display_proc (bdb)

	if(!bdb.close())
		{
		val ecode = bdb.ecode()
//		println("close error: " + bdb.errmsg(ecode))
		println("close error: ")
		}

	println ("*** 終了 ***")
}

// -------------------------------------------------------------------
}
// -------------------------------------------------------------------
